ddtrace/tracer.py

```diff
@@ -399,30 +418,27 @@ def start_span(
             root_span = tracer.start_span('web.request')
             span = tracer.start_span('web.decoder', child_of=root_span)
 
-        Or if you have a ``Context`` object::
-
-            context = tracer.get_call_context()
-            span = tracer.start_span('web.worker', child_of=context)
+        Be sure to finish all spans to avoid memory leaks and incorrect
+        parenting of spans.
         """
-        new_ctx = self._check_new_process()
+        self._check_new_process()
 
+        parent = None  # type: Optional[Span]
         if child_of is not None:
             if isinstance(child_of, Context):
-                context = new_ctx or child_of
-                parent = child_of._get_current_span()
```

Comment:
Good question!

The reason why is that we can no longer get a reference to a span when `child_of` is a context. Before, we could get it using `_get_current_span`. Note that we do still get the `parent_id` if `child_of` is a context.

The difference here (which is breaking) is that a strong span reference (`span._parent`) will not be set when `child_of` is a context.

Reply:
Just to clarify, when would we expect the `child_of` to be a context? Would that only be in distributed tracing scenarios and therefore not have a strong span reference?

Reply:
> Would that only be in distributed tracing scenarios and therefore not have a strong span reference

Bingo!

Only in context propagation cases we'd expect to get a context as a parent. It is a notable change since if you try to copy context across a thread it used to maintain a strong parent reference and now it doesn't.